src/connector/src/common.rs

```diff
@@ -684,3 +685,188 @@ impl NatsCommon {
+
+    /// Port of the mqtt broker, defaults to 1883 for tcp and 8883 for ssl
+    #[serde(rename = "port")]
+    pub port: Option<i32>,
+
+    /// The topic name to subscribe or publish to. When subscribing, it can be a wildcard topic. e.g /topic/#
+    #[serde(rename = "topic")]
+    pub topic: String,
+
+    /// Username for the mqtt broker
+    #[serde(rename = "username")]
+    pub user: Option<String>,
+
+    /// Password for the mqtt broker
+    #[serde(rename = "password")]
+    pub password: Option<String>,
+    #[serde(rename = "client_prefix")]
+
+    /// Prefix for the mqtt client id
+    pub client_prefix: Option<String>,
+
+    /// `clean_start = true` removes all the state from queues & instructs the broker
+    /// to clean all the client state when client disconnects.
+    ///
+    /// When set `false`, broker will hold the client state and performs pending
+    /// operations on the client when reconnection with same `client_id`
+    /// happens. Local queue state is also held to retransmit packets after reconnection.
+    #[serde(rename = "clean_start")]
+    #[serde(default, deserialize_with = "deserialize_bool_from_string")]
+    pub clean_start: bool,
+
+    /// The maximum number of inflight messages. Defaults to 100
+    #[serde(rename = "inflight_messages")]
+    #[serde_as(as = "Option<DisplayFromStr>")]
+    pub inflight_messages: Option<usize>,
+    #[serde(rename = "tls.ca")]
+
+    /// Path to CA certificate file for verifying the broker's key.
+    pub ca: Option<String>,
+    #[serde(rename = "tls.client_cert")]
+
+    /// Path to client's certificate file (PEM). Required for client authentication.
+    /// Can be a file path under fs:// or a string with the certificate content.
+    pub client_cert: Option<String>,
+    #[serde(rename = "tls.client_key")]
+
+    /// Path to client's private key file (PEM). Required for client authentication.
+    /// Can be a file path under fs:// or a string with the private key content.
+    pub client_key: Option<String>,
+}
+
+impl MqttCommon {
+    pub(crate) fn build_client(
+        &self,
+        id: u32,
+    ) -> ConnectorResult<(rumqttc::v5::AsyncClient, rumqttc::v5::EventLoop)> {
+        let ssl = self
+            .protocol
+            .as_ref()
+            .map(|p| p == &Protocol::Ssl)
+            .unwrap_or_default();
+
+        let client_id = format!(
+            "{}_{}{}",
+            self.client_prefix.as_deref().unwrap_or("risingwave"),
+            id,
+            std::time::SystemTime::now()
+                .duration_since(std::time::UNIX_EPOCH)
+                .unwrap()
+                .as_millis()
+                % 100000,
+        );
+
+        let port = self.port.unwrap_or(if ssl { 8883 } else { 1883 }) as u16;
+
+        let mut options = rumqttc::v5::MqttOptions::new(client_id, &self.host, port);
+        options.set_keep_alive(std::time::Duration::from_secs(10));
+
+        options.set_clean_start(self.clean_start);
+
+        if ssl {
```

Comment:
shall we move the auth part into a separate func? I am quite sure we will find a way to deal with ca cert as files on the cloud portal, cc @wyhyhyhyh